common/src/main/java/io/netty/util/internal/CleanerJava9.java

```diff
@@ -73,10 +81,35 @@ static boolean isSupported() {
 
     @Override
     public void freeDirectBuffer(ByteBuffer buffer) {
-        try {
-            INVOKE_CLEANER.invoke(PlatformDependent0.UNSAFE, buffer);
-        } catch (Throwable cause) {
-            PlatformDependent0.throwException(cause);
+        // Try to minimize overhead when there is no SecurityManager present.
+        // See https://bugs.openjdk.java.net/browse/JDK-8191053.
+        if (System.getSecurityManager() == null) {
+            try {
+                INVOKE_CLEANER.invoke(PlatformDependent0.UNSAFE, buffer);
+            } catch (Throwable cause) {
+                PlatformDependent0.throwException(cause);
+            }
+        } else {
+            freeDirectBufferPrivileged(buffer);
+        }
+    }
+
+    private static void freeDirectBufferPrivileged(final ByteBuffer buffer) {
+        Exception error = AccessController.doPrivileged(new PrivilegedAction<Exception>() {
```

Comment:
idle question: why not use PrivilegedExceptionAction ?

Reply:
I think there is not much advantage here.. We will need to unwrap then etc.